### DISCUSSION TITLE
canAccessFilament Not Being Called

### DISCUSSION THREAD

**Discussion by falconsmilie** [filamentphp/filament#6175] [Help] [answered]
Hi.

I have read the docs [here](https://filamentphp.com/docs/2.x/admin/users#authorizing-access-to-the-admin-panel) and have a User model which includes the following code;

```
use Filament\Models\Contracts\FilamentUser;

class User extends \Illuminate\Foundation\Auth\User implements FilamentUser
{
    public function canAccessFilament(): bool
    {
        return $this->role === UserRole::Admin
    }

    public function getFilamentName(): string
    {
        return "$this->first_name $this->last_name";
    }
```
but it seems `canAccessFilament()` is never being called because`\Filament\Http\Middleware\Authenticate::authenticate` is never being called. However, `\Illuminate\Auth\Middleware\Authenticate::authenticate` is being called.

Methods like `getFilamentName()` **_are_** being called.

This means regular users can access the `filament.path` admin route. The only thing preventing users from CRUD on the resources are the policies which i've enabled in `\App\Providers\AuthServiceProvider`.

Any ideas why the application would not be calling `canAccessFilament()`?

**Comment by danharrin** [comment]
It's only checked in production. So change your .env to test the behaviour

**Reply by falconsmilie** [reply]
Hi Dan. Thanks for replying. I had my env set to production on local and changing to test didn't change anything.

I updated my description to include;

`\Filament\Http\Middleware\Authenticate::authenticate` is never being called. However, `\Illuminate\Auth\Middleware\Authenticate::authenticate` is being called.

**Reply by danharrin** [reply, marked as answer]
Is `\Filament\Http\Middleware\Authenticate` middleware in your Filament config file

**Reply by falconsmilie** [reply]
Argh, that's the issue. Sorry for wasting your time. Thanks Dan.
